data/benchmarks.txt

```diff
@@ -0,0 +1,74 @@
+psql:
+select id from listings where street like '%Cartwright%' and city like '%Sophia%';
+-- benchmark: 37.501s, no way to index for this
+
+Find listing where street includes "x" and in the zip code equals y
+psql:
+select id from listings where street like '%Cartwright%' and zip = 86680; 
+-- benchmark: 18.150s
+-- after creating index on listings(zip, street): 0.101s, after creating index on listings(zip): 0.850 ms
+mongo:
+db.listings.find({street: /West/, zip: 06282})
+-- benchmark: >20 min
+
+Find listings where the city equals Griffinview
+psql:
+select * from listings where city = 'Griffinview';
+-- benchmark: 26.407s
+-- after adding index on listings(city): 1.285 ms
+mongo:
+db.listings.find({city: 'Griffinview'})
+-- benchmark: 
+
+Create new listing
+psql:
+insert into listings (street, city, state, zip, listing_agent) values ('1225 Vienna Dr', 'Sunnyvale', 'CA', '94089', '14');
+-- after adding indexes: 81.366 ms
+mongo:
+db.listings.insertOne({id: 10000003, street: "123 Fake Street", city: "Fake City", state: "CA", zip: 94105, agent: {id: 1, name: "Miss Delbert Hermann", company: "Mayer Inc", reviews: 86, recentSales: 12, phone: "(522) 705-3581", premier: true, zip: 59736, photo: "https://s3.amazonaws.com/uifaces/faces/twitter/gojeanyn/128.jpg"}})
+-- after indexes:
+
+Find all info associated with a listing
+psql:
+SELECT listings.street, listings.city, listings.state, listings.zip as home_zip,agents.name, agents.reviews, agents.recentSales, agents.phone, agents.premier, agents.photo FROM listings INNER JOIN agents on listings.listing_agent = agents.id WHERE listings.id = 9000000;
```

Comment:
Both are valid query methods, but why use caps for certain postgres queries and no caps for other postgres queries?

Reply:
no reason. this document isn't used for anything it's just my own notes